src/core/datastore/deltaset.js

```diff
@@ -137,23 +117,40 @@ function makeRequest(collectionNameWithTag, deltaSetQuery, query, options) {
     trace: options.trace,
     client
   });
-  return request.execute()
-    .then((response) => {
-      return updateDeltaSetQuery(deltaSetQuery, response).then(() => response.data);
-    })
-    .then((data) => {
-      return updateDocs(collectionNameWithTag, data);
-    });
+  return request.execute();
 }
 
-export function deltaSet(collectionNameWithTag, query, options) {
-  return getDeltaSetQuery(collectionNameWithTag, query)
-    .then((deltaSetQuery) => makeRequest(collectionNameWithTag, deltaSetQuery, query, options))
-    .then(() => readDocs(collectionNameWithTag, query))
+export function deltaSet(collectionName, query, options) {
+  return getCachedQuery(collectionName, query)
+    .then((cachedQuery) => {
+      let promise;
+
+      if (cachedQuery && cachedQuery.lastRequest) {
+        if (query && (query.skip != null || query.limit != null)) {
+          return Promise.reject(new KinveyError('You cannot use the skip and limit modifiers on the query when performing a delta set request.'));
+        }
+
+        promise = makeDeltaSetRequest(collectionName, cachedQuery.lastRequest, query, options);
+      } else {
+        promise = makeRegularGETRequest(collectionName, query, options);
+      }
+
+      return promise
+        .then((response) => {
+          const requestStartDate = response.headers.get(xKivneyRequestStartHeader);
```

Comment:
Just noticed a typo here while randomly browsing code - `xKivneyRequestStartHeader` :)

Reply:
I think we don't need the "x" in there too, while we're on the topic :)